src/components/Heading/Heading.tsx

```diff
@@ -1,56 +1,82 @@
+  'alert',
+  'base',
+  'brand',
+  'inherit',
+  'neutral',
+  'success',
+  'warning',
+  'white',
   /**
-   * Child node(s) that can be nested inside component
-   */
-  children: ReactNode;
+   * @deprecated Info variant is deprecated.
+   */ 'info',
+] as const;
+export type Variant = typeof VARIANTS[number];
+export type HeadingElement = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
+
+type Props = {
   /**
-   * CSS class names that can be appended to the component.
+   * This prop can be used to specify which size heading should
+   * actually be rendered, in the case that you want to render an element
+   * as one heading but style it as if it were another. If both an `as` prop
+   * and a `size` prop are passed, the `as` will be used to determine the element
+   * and the `size` will be used to determine the styling.
    */
+  as?: HeadingElement;
+  children: React.ReactNode;
   className?: string;
-  /**
-   * Id of the heading
-   */
-  id?: string;
-  /**
-   * Size variants of the Heading component, ranging from 1-7. These numbers map to typographic sizing/style presets. This prop is used in instances where a Heading needs to use a specific tag (such as `h3`) for semantic reasons, but needs to be styled like something other than its default styles (such as an `h2` or `h1`).
-   */
-  size?: 1 | 2 | 3 | 4 | 5 | 6;
-  /**
-   * The rendered tag name of the Heading. The tag name should always relate to the [document outline](http://html5doctor.com/outlines/) and a tag name should never be chosen for its default aesthetic qualities. If a specific style is desired, use the `size` prop to manipulate the Heading style.
-   */
-  as: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
-}
+  size: HeadingElement;
+  tabIndex?: number;
+  variant?: Variant;
+} & React.HTMLAttributes<HTMLHeadingElement>;
 
 /**
- * Primary UI component for user interaction
+ * ```ts
+ * import {Heading} from "@chanzuckerberg/eds-components";
+ * ```
  */
-export const Heading = ({
-  className,
-  children,
-  as,
-  size,
-  id,
-  ...other
-}: Props) => {
-  const componentClassName = clsx(
-    styles['heading'],
-    className,
-    size === 1 && styles['heading--size-1'],
-    size === 2 && styles['heading--size-2'],
-    size === 3 && styles['heading--size-3'],
-    size === 4 && styles['heading--size-4'],
-    size === 5 && styles['heading--size-5'],
-    size === 6 && styles['heading--size-6'],
-  );
 
-  const TagName = as;
-  return (
-    <TagName className={componentClassName} {...other}>
-      {children}
-    </TagName>
-  );
-};
+export const Heading = forwardRef(
+  (
+    {
+      as,
+      children,
+      className,
+      variant,
+      size,
+      /**
+       * Components that wrap typography sometimes require props such as
+       * event handlers, tabIndex, etc. to be passed down into the element.
+       * TODO: add better typing or documentation for optional props,
+       * e.g. React.HTMLAttributes<HTMLHeadingElement>
```

Comment:
I wonder if this todo comment is still relevant when we're letting all components pass down `other` now.

Reply:
Hmm you're right, I'll change the comment to solely why we're passing other and rename `rest` to `other`, thanks